### PR TITLE
stripe-node v7.0.0

### PR DESCRIPTION
**Pull request by rattrayalex-stripe**
This is a major release of stripe-node, dropping support for several old versions of Node and deprecated methods.

[The migration guide](https://github.com/stripe/stripe-node/wiki/Migration-guide-for-v7) contains a detailed list of backwards-incompatible changes with upgrade instructions.

cc @stripe/api-libraries 
cc @ob-stripe @paulasjes-stripe @robz-stripe 

⚠️ = breaking changes

- [x] ⚠️ Drop support for Node 4, 5 and 7 (#606)
- [x] Prettier formatting (#604)
- [x] Alphabetize “basic” methods (#610)
- [x] Use `id` for single positional arguments (#611)
- [x] Modernize ES5 to ES6 with lebab (#607)
- [x] ⚠️ 🔥Remove deprecated methods 🔥 (#613)
- [x] Add VSCode and EditorConfig files (#620)
- [x] ⚠️ Drop support for Node 9 and bump dependencies to latest versions (#614)
- [x] Misc. manual formatting (#623)
- [x] ⚠️ Remove legacy parameter support in `invoices.retrieveUpcoming()` (#621)
- [x] ⚠️ Remove curried urlData and manually specified urlParams (#625)
- [x] Extract resources file (to aid with code generation) (#626)

### follow-on 

- [ ] Removed most instances of `self` (#615)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "stripe",
-  "version": "6.31.2",
+  "version": "7.0.0",
```
Comment (ob-stripe) [inline package.json]: No need to change that, the release script will handle it.

Comment (rattrayalex-stripe) [reply package.json]: Can I change it anyway? makes intent more clear

Comment (ob-stripe) [reply package.json]: Sure, it might just be annoying if the v7 branch is long-lived and we release more 6.x versions since it will cause merge conflicts.

**Comment by ob-stripe** [comment]
@rattrayalex-stripe @irace-stripe We try to keep the commit history as clean as possible in the libraries repos. Do you mind squashing the commits here? I think there should only be two so far in this branch:
- one for dropping support for Node 4/5/7
- one for Prettier formatting changes

**Comment by irace-stripe** [comment]
@ob-stripe @rattrayalex-stripe This branch has now been rebased to have a clean commit history.

**Comment by irace-stripe** [comment]
Added another branch off of this one: https://github.com/stripe/stripe-node/pull/610

**Comment by ob-stripe** [comment]
@rattrayalex-stripe @irace-stripe You should rebase on `master` to get the new `balanceTransactions` resource.

**Comment by rattrayalex-stripe** [comment]
I might actually wait to rebase master till we're removed deprecated things too, so we can rebase once just before merging/releasing.

**Comment by rattrayalex-stripe** [comment]
Alright, I think this is ready to go!

cc @ob-stripe I intend to merge and release within an hour or two.

**Pull request merged**